emcc.py

```diff
@@ -1937,6 +1937,9 @@ def is_link_flag(flag):
         return any(flag.startswith(x) for x in ('-l', '-L', '-Wl,'))
 
       compile_args = [a for a in newargs if a and not is_link_flag(a)]
+      if '-fPIC' in compile_args and not shared.Settings.RELOCATABLE:
+        shared.warning('ignoring -fPIC flag when not building with SIDE_MODULE or MAIN_MODULE')
+        compile_args.remove('-fPIC')
```

Comment:
This change means that people need to compile with `SIDE/MAIN_MODULE` and not just link? Is that new or am I forgetting how upstream worked here?

Reply:
Yes, I think that is true.  But they will see this warning so it should be clear.

Reply:
I think this is pretty good intermediate fix until the binaryen fix is ready.

Reply:
Sounds good.

How do our current tests pass, then? It seems like all dylink/dlfcn tests should warn if they include more than 1 file, since I don't see this flag added to their compile commands?

Reply:
Agreed it's good to have a temp fix. I'm just trying to fully understand it.

Reply:
But none of our dylink currently pass `-fPIC`.   So this means they must all be passing `SIDE/MAIN_MODULE` during both compile and link.

As far as I know we never advertised the fact that `-fPIC` could be used at compile time instead of `SIDE/MAIN_MODULE`.

Reply:
Interesting, good point.

We should probably add more realistic tests then, if users are using `-fPIC` for dynamic linking but our tests aren't.

Reply:
IIUC users are not using `fPIC` for dyamic linking, they are using it simply because their build system always adds it in, but they are doing static linking.